library/std/src/panic.rs

```diff
@@ -316,16 +316,18 @@ where
 ///
 /// # Notes
 ///
-/// Note that this function **might not catch all panics** in Rust. A panic in
-/// Rust is not always implemented via unwinding, but can be implemented by
-/// aborting the process as well. This function *only* catches unwinding panics,
-/// not those that abort the process.
-///
-/// Note that if a custom panic hook has been set, it will be invoked before
-/// the panic is caught, before unwinding.
-///
-/// Also note that unwinding into Rust code with a foreign exception (e.g.
-/// an exception thrown from C++ code) is undefined behavior.
+/// This function **might not catch all panics** in Rust. A panic in Rust is not
+/// always implemented via unwinding, but can be implemented by aborting the
+/// process as well. This function *only* catches unwinding panics, not those
+/// that abort the process.
+///
+/// If a custom panic hook has been set, it will be invoked before the panic is
+/// caught, before unwinding.
+///
+/// Although unwinding into Rust code with a foreign exception (e.g. an
+/// exception thrown from C++ code) via an appropriate ABI (e.g. `"C-unwind"`)
```

Comment:
One could argue that there should be some indication that this is only well-defined as of Rust 1.71, but prior to that there was no "appropriate ABI", and mentioning `"C-unwind"` reinforces this, so I think that explicitly mentioning Rust 1.71 would be redundant.